common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1155,6 +1156,71 @@ public UTF8String translate(Map<String, String> dict) {
     return fromString(sb.toString());
   }
 
+  public UTF8String translate(Map<String, String> dict, int collationId) {
+    if(CollationFactory.fetchCollation(collationId).supportsBinaryEquality) {
+      return translate(dict);
+    }
+    return collationAwareTranslate(dict, collationId);
+  }
+
+  public UTF8String collationAwareTranslate(Map<String, String> dict, int collationId) {
+    if (numBytes == 0) {
+      return this;
```

Comment:
I don't see this behaviour in regular translate. So won't this mean that regular translate return a new string and collation aware one would return the same instance if called on an empty string? Probably not what we want.